src/executionFiles/bridges/nxtp.execute.ts

```diff
@@ -69,10 +82,7 @@ export class NXTPExecutionManager {
       if (!this.shouldContinue) return status
       // -> request key
       try {
-        const encryptionPublicKey = await (window as any).ethereum.request({
-          method: 'eth_getEncryptionPublicKey',
-          params: [await signer.getAddress()], // you must have access to the specified account
-        })
+        const encryptionPublicKey = await hooks.getPublicKeyHook()
```

Comment:
Before executing the whole encryption process, we should check if a `getPublicKeyHook` is defined or not. Because if there is no hook we can skip the whole process and also not show it in the UI. The default `getPublicKeyHook` has to be undefined to make this work